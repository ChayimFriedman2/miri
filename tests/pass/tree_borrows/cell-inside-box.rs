//@compile-flags: -Zmiri-tree-borrows
#![feature(box_as_ptr)]
#[path = "../../utils/mod.rs"]
#[macro_use]
mod utils;

use std::cell::UnsafeCell;

pub fn main() {
    let cell = UnsafeCell::new(42);
    let mut root = Box::new(cell);

    let a = Box::as_mut_ptr(&mut root);
    unsafe {
        name!(a);
        let alloc_id = alloc_id!(a);
        print_state!(alloc_id);
    }
}
