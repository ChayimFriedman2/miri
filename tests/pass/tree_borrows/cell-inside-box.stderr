──────────────────────────────────────────────────
Warning: this tree is indicative only. Some tags may have been hidden.
0..   4
| Act |    └─┬──<TAG=root of the allocation>
| ReIM|      └────<TAG=a>
──────────────────────────────────────────────────
